shared/chat/inbox/row/teams-divider/container.tsx

```diff
@@ -13,23 +13,17 @@ type OwnProps = {
   style?: StylesCrossPlatform
 }
 
-const mapStateToProps = state => ({_badges: state.chat2.badgeMap, _metaMap: state.chat2.metaMap})
+const mapStateToProps = (state: TypedState) => ({
+  _badges: state.chat2.badgeMap,
+  _inboxLayout: state.chat2.inboxLayout,
+})
 
-const getMetaCounts = memoize((badges, metaMap) => {
+const getMetaCounts = memoize((badges, inboxLayout: RPCChatTypes.UIInboxLayout | null) => {
   let badgeCount = 0
   let hiddenCount = 0
-  metaMap.forEach(meta => {
-    if (meta.teamType === 'big') {
-      return
-    }
-    if (meta.status === RPCChatTypes.ConversationStatus.ignored) {
-      return
-    }
-    const id = meta.conversationIDKey
-    if (!Constants.isValidConversationIDKey(id)) {
-      return
-    }
-
+  const smallTeams = inboxLayout ? inboxLayout.smallTeams || [] : []
+  smallTeams.forEach((conv: RPCChatTypes.UIInboxSmallTeamRow) => {
```

Comment:
it'd be nice if this was just sent down the pipe

Reply:
The badge state is intentionally separate from this, I'm hesitant to have it also tacked onto the layout as well for now.